timemachine/cpp/src/barostat.cu

```diff
@@ -140,31 +142,31 @@ void MonteCarloBarostat::inplace_move(
         return;
     }
 
-    curandErrchk(curandSetStream(cr_rng_, stream));
-    // Generate scaling and metropolis conditions in one pass
-    curandErrchk(curandGenerateUniformDouble(cr_rng_, d_rand_, 2));
+    // Get offset into the d_rand_ array
+    int random_offset = (((step_ / interval_) * 2) - 2) % (RANDOM_BATCH_SIZE * 2);
```

Comment:
This needs a bit more documentation - if `step_ == 0` doesn't this result in a `(-2 % RANDOM_BATCH_SIZE*2)`?

Reply:
`step_` can't equal to zero as `interval_ > 0` and this code only runs if `step_ % interval_ == 0`